analyses/chromosomal-instability/util/chr-break-calculate.R

```diff
@@ -0,0 +1,233 @@
+  #   start_col: character string that indicates the column name with the
+  #              start coordinate. Default is "start".
+  #   end_col: character string that indicates the column name with the
+  #            end coordinate. Default is "end".
+  #
+  # Returns:
+  # A Genomic Ranges formatted object for the particular sample that has the
+  # break points from the data.frame
+
+  # If no data.frame is provided, stop
+  if (is.null(break_df)) {
+    stop("No breaks data.frame has been provided. ")
+  }
+
+  # List the columns we need
+  needed_cols <- c(samples_col, chrom_col, start_col, end_col)
+
+  # Get logical vector indicating which are in metadata
+  found_cols <- (needed_cols %in% colnames(break_df))
+
+  # If not all the columns are found, stop
+  if (!all(found_cols)) {
+    stop(cat(
+      "The following column names specified for making the GenomicRanges object: \n",
+      paste(needed_cols[which(!found_cols)], collapse = "\n"),
+      "\n ...were not found in the specified metadata data.frame.",
+      "Check your `_col` arguments."
+    ))
+  }
+
+  if (sample_id[1] != "all") {
+    # Extract samples' info from the data.frame
+    break_df <- break_df %>%
+      dplyr::filter(!!rlang::sym(samples_col) %in% c(sample_id))
+
+    # Stop if the sample doesn't exist in the data.frame
+    if (nrow(break_df) == 0) {
+      stop("No sample data was found for the given `sample_id` argument.")
+    }
+  }
+
+  # Make GRanges for CNV data
+  granges <- GenomicRanges::GRanges(
+    seqnames = dplyr::pull(break_df, !!rlang::sym(chrom_col)),
+    ranges = IRanges::IRanges(
+      start = dplyr::pull(break_df, !!rlang::sym(start_col)),
+      end = dplyr::pull(break_df, !!rlang::sym(end_col))
+    ),
+    mcols = dplyr::pull(break_df, !!rlang::sym(samples_col))
+  )
+
+  return(granges)
+}
+
+break_density <- function(sv_breaks = NULL,
+                          cnv_breaks = NULL,
+                          sample_id = NULL,
+                          window_size = 1e6,
+                          chr_sizes_list = NULL,
+                          samples_col_cnv = "samples",
+                          chrom_col_cnv = "chrom",
+                          start_col_cnv = "start",
+                          end_col_cnv = "end",
+                          samples_col_sv = "samples",
+                          chrom_col_sv = "chrom",
+                          start_col_sv = "start",
+                          end_col_sv = "end") {
+  # For given breaks data.frame(s), calculate the breaks density for a tiled
+  # windows across the genome. Returns the data as a GenomicRanges object for
+  # easy mapping with `ggbio`. Where the density and counts are stored in
+  # @elementMetadata@listData.
+  #
+  # Args:
+  #   sv_breaks: a data.frame with the breaks for the SV data.
+  #   cnv_breaks: a data.frame with the breaks for the SV data.
+  #   sample_id: sample ID for which the data needs to be extracted and made into
```

Comment:
A bit more clarity here: My immediate intuition is that this refers literally to `sample_id` column in other files. If so, say explicitly it's the same as in histology file.

Reply:
The `sample_ID` argument can literally be any kind of sample ID, so long as the column specified in with `sample_col` arguments  for `sv_breaks` and `cnv_breaks` have a string that matches it.